scala-thoughts.md

```diff
@@ -0,0 +1,29 @@
+###### What is easy to do in Scala? What's not?
+Easy:
+- String concatenation
+    - Strings can be added to other strings, and strings can be concatenated from a List form.
+- Type checking/aliasing
+    - I love static typing - especially when running Scala in codespaces can cause issues. I also find it cool that you can name a custom type (ex. Color, Board in Mastermind) and I can see how this would help with  readability of objects in domain-specific programs.
+
+Hard:
+- List indexing
+    - I assumed "apply" does some sort of transform to the list, and only realized apply meant get element by index when I stumbled upon the shorthand for indexing in Mastermind.
```

Comment:
This is really interesting and something I didn't notice, but I would have also assumed it was a transform and not a way to get elements from a list.